syncGraph.py

```diff
@@ -0,0 +1,276 @@
+    '''
+
+    vprint(f'Loading in graph from "{fpath}"')
+
+    # Grab JSON
+    with open(fpath) as f:
+        data = json.load(f)
+
+    posts = data['posts']
+    edges = data['edges']
+
+    return posts, edges
+
+
+def getSubimageWeightsMeta(edges, source, target):
+    '''
+    @desc  Returns subimage weight (float, else None) and metadata (string, else None)
+    '''
+    sub_edges = edges[source][target].get('subimage', [])
+    labels = []
+    sw = 0 if len(sub_edges) > 0 else None
+    sm = None
+    if len(sub_edges) > 0:
+        for tup in sub_edges:
+            sw += tup[0]
+            labels.append(tup[1])
+        sm = '|'.join(labels)
+    return sw, sm
+
+
+def getRelTxtWeightsMeta(posts, edges, source, target):
+    '''
+    @desc  Returns rel-text weight (float, else None) and metadata (string, else None)
+    '''
+    rw = edges[source][target].get('rel_text')
+    rm = None
+    if rw:
+        rm = eval(posts[source].get('related_text', 'set()')).intersection(eval(posts[target].get('related_text', 'set()')))
+        rm = '|'.join(rm)
+    return rw, rm
+
+
+def getOcrWeightsMeta(posts, edges, source, target):
+    '''
+    @desc  Returns ocr weight (float, else None) and metadata (string, else None)
+    '''
+    ow = edges[source][target].get('ocr')
+    om = None
+    if ow:
+        om = eval(posts[source].get('ocr', 'set()')).intersection(eval(posts[target].get('ocr', 'set()')))
+        om = '|'.join(om)
+    return ow, om
+
+
+def insertPostRelatedness(cursor, source, target, rw, rm, sw, sm, ow, om):
+    '''
+    @desc   Insert information into Post Relatedness
+    --
+    @param  cursor  cursor for mysql.connector
+    @param  source  scrape_id of post 1
+    @param  target  scrape_id of post 2
+    @param  rw      related text weight
+    @param  rm      related text metadata
+    @param  sw      subimage weight
+    @param  sm      subimage metadata
+    @param  ow      ocr weight
+    @param  om      ocr metadata
+    '''
+
+    # Query Structure
+    sql = '''
+    INSERT INTO mews_app.PostRelatedness (
+        post1_id,
+        post2_id,
+        rel_txt_wt, rel_txt_meta, 
+        sub_img_wt, sub_img_meta, 
+        ocr_wt, ocr_meta
+    )
+    VALUES (
+        (SELECT id FROM mews_app.Posts WHERE scrape_id = %(source)s),
+        (SELECT id FROM mews_app.Posts WHERE scrape_id = %(target)s),
+        %(rw)s, %(rm)s, 
+        %(sw)s, %(sm)s,
+        %(ow)s, %(om)s
+    )
+    ;
+    '''
+
+    vprint(f'Inserting Source: {source}; Target: {target}; Weights (r-s-o): {rw}-{sw}-{ow}')
+    vprint(f'  R Meta: {rm}')
+    vprint(f'  S Meta: {sm}')
+    vprint(f'  O Meta: {om}')
+
+    # Query Arguments
+    args = {
+        'source': source,
+        'target': target,
+        'rw': rw,
+        'rm': rm,
+        'sw': sw,
+        'sm': sm,
+        'ow': ow,
+        'om': om
+    }
+
+    # Run Query
+    try:
+        cursor.execute(sql, args)  
+    except mysql.connector.Error:
+        raise
```

Comment:
What is the purpose of the except/raise?

Reply:
Re-raising the exception so that line 246 can handle it appropriately.